tools/cli/main.c

```diff
@@ -106,10 +106,29 @@ int main(int argc, char **argv)
 
         if (pCmd)
         {
-            if (pCmd->ReqRoot && geteuid())
+            if (pCmd->ReqRoot)
             {
-                dwError = ERROR_TDNF_PERM;
-                BAIL_ON_CLI_ERROR(dwError);
+                if (geteuid())
+                {
+                    dwError = ERROR_TDNF_PERM;
+                    BAIL_ON_CLI_ERROR(dwError);
+                }
+
+                /*
+                 * For operations other than makecache & clean
+                 * suppressing info and prompting y/n doesn't make sense
+                 * Enable info if assumeyes or assumeno is not provided
+                 * along with quiet.
+                 */
+                if ((strcmp(pszCmd, "makecache") &&
+                     strcmp(pszCmd, "clean")) &&
+                     pCmdArgs->nQuiet &&
+                     !pCmdArgs->nAssumeYes && !pCmdArgs->nAssumeNo)
+                {
+                    pr_err("--assumeyes|-y or --assumeno is must when quiet is set\n");
+                    dwError = ERROR_TDNF_CLI_NOT_ENOUGH_ARGS;
```

Comment:
I think it's better to add a new error code. This would print "The command line parser could not continue. Expected at least one argument.", which is confusing.

In that case, you also don't need the `pr_err()`, because you can set the error code to "--assumeyes|-y or --assumeno is required when quiet is set".

Reply:
Agree